### ISSUE TITLE
What do the metrics wps, ups and wpb mean in the training logger ?

### ISSUE THREAD

**Issue by shahbazsyed** [pytorch/fairseq#2201]
In the following dictionary from the training log output:

<pre>{&quot;epoch&quot;: 27, &quot;update&quot;: 26.267, &quot;loss&quot;: &quot;8.206&quot;, &quot;nll_loss&quot;: &quot;7.049&quot;, &quot;ppl&quot;: &quot;132.47&quot;, &quot;wps&quot;: &quot;1195.4&quot;, &quot;ups&quot;: &quot;1.62&quot;, &quot;wpb&quot;: &quot;738.1&quot;, &quot;bsz&quot;: &quot;46.4&quot;, &quot;num_updates&quot;: &quot;33700&quot;, &quot;lr&quot;: &quot;0.00017226&quot;, &quot;gnorm&quot;: &quot;1.833&quot;, &quot;clip&quot;: &quot;1&quot;, &quot;train_wall&quot;: &quot;61&quot;, &quot;wall&quot;: &quot;30542&quot;}</pre>

I assume the following from looking at the code and other issues:
**bsz** = batch size 
**gnorm** = L2 norm of the gradients
**clip** = gradient clipping threshold
**train_wall** = time taken for one training step
**wall** = total time spent training, validating, saving checkpoints (so far)
**wps** = ?
**ups** = ?
**wpb** = ?

**Comment by gvskalyan** [comment]
wps - Words Per Second
ups - Updates Per Second
wpb - Words Per Batch

**Comment by shahbazsyed** [comment]
Thanks!

**Comment by benjamin3344** [comment]
Anyone know what nvo, stp is short for? And what does the "words" mean in wps and wpb..  @gvskalyan @shahbazsyed
